### PR TITLE
Add logs for local echo issues

### PR DESCRIPTION
**Pull request by ismailgulek**
This PR adds logs to `MXKRoomDataSource` to better understand vector-im/element-ios#1771

Fixes vector-im/element-ios#4822

Requires https://github.com/matrix-org/matrix-ios-sdk/pull/1225

### CODE REVIEW

File: MatrixKit/Models/Room/MXKRoomDataSource.m [hunk 1]
```diff
@@ -2387,6 +2392,8 @@ - (void)replaceEvent:(MXEvent*)eventToReplace withEvent:(MXEvent*)event
 {
     NSMutableArray *deletedRows = [NSMutableArray array];
     
+    MXLogVerbose(@"[MXKRoomDataSource][%p] removeCellData: %@", self, cellData.events);
```
Comment (manuroe) [inline MatrixKit/Models/Room/MXKRoomDataSource.m:2395]: There is a risk to leak events content.

File: MatrixKit/Models/Room/MXKRoomDataSource.m [hunk 2]
```diff
@@ -2557,6 +2571,8 @@ - (void)localEventDidChangeIdentifier:(NSNotification *)notif
     
     if (event && previousId)
     {
+        MXLogVerbose(@"[MXKRoomDataSource][%p] localEventDidChangeIdentifier from: %@ to: %@", self, previousId, event.eventId);
```
Comment (manuroe) [inline MatrixKit/Models/Room/MXKRoomDataSource.m:2574]: We can have this log before the `if` in case the notif to track malformed notif.

**Review by manuroe — COMMENTED** [review]
We could have a log in the entry of  `eventDidChangeSentState` like you did for localEventDidChangeIdentifier

File: MatrixKit/Models/Room/MXKRoomDataSource.m (hunk 1 above)
Comment (ismailgulek) [reply MatrixKit/Models/Room/MXKRoomDataSource.m]: Updated to log only _eventId_.

File: MatrixKit/Models/Room/MXKRoomDataSource.m (hunk 2 above)
Comment (ismailgulek) [reply MatrixKit/Models/Room/MXKRoomDataSource.m]: Moved

**Comment by ismailgulek** [comment]
> We could have a log in the entry of `eventDidChangeSentState` like you did for localEventDidChangeIdentifier

Added

File: MatrixKit/Controllers/MXKRoomViewController.m [hunk 3]
```diff
@@ -380,6 +380,9 @@ - (void)viewDidAppear:(BOOL)animated
     {
         hasAppearedOnce = YES;
     }
+    
+    //  Mark all messages as read when the room is displayed
+    [self.roomDataSource.room.summary markAllAsRead];
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKRoomViewController.m:385]: This is too brutal. It will reset the read marker as well.

Comment (manuroe) [reply MatrixKit/Controllers/MXKRoomViewController.m:385]: It seems we need a new method in MXRoomSummary to reset local unread counts without changing the read marker.

Comment (ismailgulek) [reply MatrixKit/Controllers/MXKRoomViewController.m]: Updated to use `markAllAsReadLocally` method.

**Review by manuroe — APPROVED** [review]

**Pull request merged**